File: CS/PersonalOrganizer/ViewModels/ContactCollectionViewModel.cs
using PersonalOrganizer.Common.DataModel;
using PersonalOrganizer.Common.ViewModel;
using PersonalOrganizer.ContactContextDataModel;
using PersonalOrganizer.Model;

namespace PersonalOrganizer.ViewModels {
    public partial class ContactCollectionViewModel : CollectionViewModel<Contact, int> {
        readonly IUnitOfWorkFactory unitOfWorkFactory;

        public ContactCollectionViewModel()
            : this(UnitOfWorkSource.GetUnitOfWorkFactory()) {
        }
        public ContactCollectionViewModel(IUnitOfWorkFactory unitOfWorkFactory) {
            this.unitOfWorkFactory = unitOfWorkFactory;
        }
        protected override IReadOnlyRepository<Contact> GetRepository() {
            return unitOfWorkFactory.CreateUnitOfWork().Contacts;
        }
    }
}

File: VB/PersonalOrganizer/ViewModels/ContactCollectionViewModel.vb
Imports System
Imports System.Linq
Imports Common.Utils
Imports ContactContextDataModel
Imports Common.DataModel
Imports PersonalOrganizer.Model
Imports Common.ViewModel
Namespace PersonalOrganizer.ViewModels
    Partial Public Class ContactCollectionViewModel
        Inherits CollectionViewModel(Of Contact, Integer)
        Private ReadOnly _unitOfWorkFactory As IUnitOfWorkFactory
        Public Sub New()
            MyClass.New(UnitOfWorkSource.GetUnitOfWorkFactory())
        End Sub
        Public Sub New(ByVal unitOfWorkFactory As IUnitOfWorkFactory)
            Me._unitOfWorkFactory = unitOfWorkFactory
        End Sub
        Protected Overrides Function GetRepository() As IReadOnlyRepository(Of Contact)
            Return _unitOfWorkFactory.CreateUnitOfWork().Contacts
        End Function
    End Class
End Namespace
